pkg/objstore/filesystem/filesystem.go

```diff
@@ -54,13 +54,10 @@ func NewBucket(rootDir string) (*Bucket, error) {
 
 // Iter calls f for each entry in the given directory. The argument to f is the full
 // object name including the prefix of the inspected directory.
-func (b *Bucket) Iter(ctx context.Context, dir string, f func(string) error) error {
+func (b *Bucket) Iter(_ context.Context, dir string, f func(string) error) error {
 	absDir := filepath.Join(b.rootDir, dir)
 	info, err := os.Stat(absDir)
 	if err != nil {
-		if os.IsNotExist(err) {
-			return nil
-		}
```

Comment:
Can you please elaborate more on this change,  is the original logic wrong or it causes bugs?

Reply:
The circle CI test failure is caused by this change.

Reply:
Just felt wrong TBH I though it will be more verbose what's wrong if there is no directory. But ok (: reverted